tools/autograd/gen_variable_type.py

```diff
@@ -658,6 +662,15 @@ def emit_any_requires_grad() -> List[str]:
         return [SETUP_ANY_REQUIRES_GRAD.substitute(
             args_with_derivatives=[arg.name for arg in args_with_derivatives]), ]
 
+    def emit_assert_no_inference_tensor() -> List[str]:
+        tensor_arg_names = []
+        for arg in f.func.arguments.tensor_args:
```

Comment:
Does that include TensorList?

Given the code below it looks like no but it should.

Can we add a test where we use `cat()` and make sure this code is properly triggered?

Reply:
Yup tensor args already include TensorList and added a `cat()` test case :D